addon/components/power-select.js

```diff
@@ -402,14 +404,43 @@ export default Component.extend({
 
   // Tasks
   triggerTypingTask: task(function* (e) {
+    // In general, a user doing this interaction means to have a different result.
+    let searchStartOffset = 1;
     let publicAPI = this.get('publicAPI');
     let charCode = e.keyCode;
     if (this._isNumpadKeyEvent(e)) {
       charCode -= 48; // Adjust char code offset for Numpad key codes. Check here for numapd key code behavior: https://goo.gl/Qwc9u4
     }
-    let term = publicAPI._expirableSearchText + String.fromCharCode(charCode);
-    this.updateState({ _expirableSearchText: term });
-    let matches = this.filter(publicAPI.options, term, true);
+    let term;
+
+    // Check if user intends to cycle through results. _repeatingChar can only be the first character.
+    let c = String.fromCharCode(charCode);
+    if (c === publicAPI._repeatingChar) {
+      term = c;
+    } else {
+      term = publicAPI._expirableSearchText + c;
+    }
+
+    if (term.length > 1) {
+      // If the term is longer than one char, the user is in the middle of a non-cycling interaction
+      // so the offset is just zero (the current selection is a valid match).
+      searchStartOffset = 0;
+      this.updateState({ _repeatingChar: '' });
+    } else {
+      this.updateState({ _repeatingChar: c });
+    }
+
+    // When the select is open, the "selection" is just highlighted.
+    if (publicAPI.isOpen && publicAPI.highlighted) {
+      searchStartOffset += indexOfOption(publicAPI.options, publicAPI.highlighted);
+    } else if (!publicAPI.isOpen && publicAPI.selected) {
+      searchStartOffset += indexOfOption(publicAPI.options, publicAPI.selected);
+    }
+
+    // The char is always appended. That way, searching for words like "Aaron" will work even
```

Comment:
The only suggestion on the code is trying to batch changes and make a single call to `this.updateState` if possible. In this case it seems that you could do it.